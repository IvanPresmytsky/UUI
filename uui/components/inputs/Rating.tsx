import { Overwrite, withMods } from '@epam/uui-core';
import * as uuiComponents from '@epam/uui-components';
import { Tooltip } from '../overlays';
import { settings } from '../../settings';

import { ReactComponent as FilledStarIcon } from '../../icons/star-filled.svg';

import css from './Rating.module.scss';

interface RatingMods {
    /**
     * Defines component size.
     * @default '18'
     */
    size?: 18 | 24 | 30;
}
export interface RatingModsOverride {}

/** Represents the 'Core properties' for the Rating component. */
export type RatingCoreProps = Omit<uuiComponents.RatingProps, 'filledStarIcon' | 'emptyStarIcon' | 'renderRating' | 'from' | 'to'>;

/** Represents the properties for the Rating component. */
export interface RatingProps extends RatingCoreProps, Overwrite<RatingMods, RatingModsOverride> {}

function applyRatingMods(mods: RatingProps) {
    return [
        css.root,
        `uui-size-${mods.size || settings.sizes.defaults.rating}`,
    ];
}

export const Rating = withMods<uuiComponents.RatingProps, RatingProps>(
    uuiComponents.Rating,
    applyRatingMods,
    () => ({
        filledStarIcon: FilledStarIcon,
        emptyStarIcon: FilledStarIcon,
        Tooltip,
    }),
);
