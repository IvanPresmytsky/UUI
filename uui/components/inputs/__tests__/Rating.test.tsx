import React from 'react';
import { Rating } from '../Rating';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('Rating', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<Rating value={ null } onValueChange={ jest.fn } />);

        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with more props', async () => {
        const tree = await renderSnapshotWithContextAsync(<Rating value={ null } onValueChange={ jest.fn } size={ 24 } step={ 1 } />);

        expect(tree).toMatchSnapshot();
    });
});
