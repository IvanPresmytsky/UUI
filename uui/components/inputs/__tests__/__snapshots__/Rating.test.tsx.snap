// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should be rendered correctly 1`] = `
<div
  aria-valuemax={5}
  aria-valuemin={1}
  aria-valuenow={0}
  className="container root uui-size-18"
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  role="slider"
  tabIndex={0}
>
  <div
    className="container"
  >
    <div
      className="uui-rating-empty-star-container"
    >
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
    </div>
    <div
      className="uui-rating-filled-star-container"
      style={
        Object {
          "width": 0,
        }
      }
    >
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Rating should be rendered correctly with more props 1`] = `
<div
  aria-valuemax={5}
  aria-valuemin={1}
  aria-valuenow={0}
  className="container root uui-size-24"
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  role="slider"
  tabIndex={0}
>
  <div
    className="container"
  >
    <div
      className="uui-rating-empty-star-container"
    >
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
    </div>
    <div
      className="uui-rating-filled-star-container"
      style={
        Object {
          "width": 0,
        }
      }
    >
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Rating should be rendered with other icon 1`] = `
<div
  aria-valuemax={5}
  aria-valuemin={1}
  aria-valuenow={0}
  className="container root uui-size-24"
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  role="slider"
  tabIndex={0}
>
  <div
    className="container"
  >
    <div
      className="uui-rating-empty-star-container"
    >
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
    </div>
    <div
      className="uui-rating-filled-star-container"
      style={
        Object {
          "width": 0,
        }
      }
    >
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
      <div
        className="uui-rating-star"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
